src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedFieldSymbolBase.cs

```diff
@@ -44,6 +44,7 @@ internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, r
             base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
 
             CSharpCompilation compilation = this.DeclaringCompilation;
+            var type = this.TypeWithAnnotations;
```

Comment:
>type [](start = 16, length = 4)

nit: `typeWithAnnotations` would be better name to avoid confusion with `this.Type` below. Consider declaring a local `var type = typeWithAnnotations.Type` to replace `this.Type`.